### ISSUE TITLE
A resource can be made unavailable by marking it trusted

### ISSUE THREAD

**Issue by JamesWTruher** [PowerShell/PSResourceGet#1302]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

```PowerShell
PS> find-psresource -name Takeown -Repository PSGallery

Name    Version Prerelease Repository Description
----    ------- ---------- ---------- -----------
Takeown 1.0.2              PSGallery  Crescendo Powershell wrapper of takeown.exe

PS> set-psresourceRepository -name PSGallery -Trusted      
PS> find-psresource -name Takeown -Repository PSGallery
Find-PSResource: Repository 'PSGallery' is not a known repository type that is supported. Please file an issue for support at https://github.com/PowerShell/PSResourceGet/issues
PS> set-PSResourceRepository -name PSGallery -ApiVersion v2
PS> find-psresource -name Takeown -Repository PSGallery    

Name    Version Prerelease Repository Description
----    ------- ---------- ---------- -----------
Takeown 1.0.2              PSGallery  Crescendo Powershell wrapper of takeown.exe
```

### Expected behavior

```console
the repository should be available
```

### Actual behavior

```console
Find-PSResource: Repository 'PSGallery' is not a known repository type that is supported. Please file an issue for support at https://github.com/PowerShell/PSResourceGet/issues
```

### Error details

```console
S> gerr                    

Exception             : 
    Type        : System.Management.Automation.PSInvalidOperationException
    ErrorRecord : 
        Exception             : 
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : Repository 'PSGallery' is not a known repository type that is supported. Please file an issue for support at 
https://github.com/PowerShell/PSResourceGet/issues
            HResult : -2146233087
        CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : InvalidOperation
    Message     : Repository 'PSGallery' is not a known repository type that is supported. Please file an issue for support at 
https://github.com/PowerShell/PSResourceGet/issues
    HResult     : -2146233079
TargetObject          : Microsoft.PowerShell.PSResourceGet.Cmdlets.FindHelper
CategoryInfo          : InvalidArgument: (Microsoft.PowerShel….Cmdlets.FindHelper:FindHelper) [Find-PSResource], PSInvalidOperationException
FullyQualifiedErrorId : RepositoryApiVersionUnknown,Microsoft.PowerShell.PSResourceGet.Cmdlets.FindPSResource
InvocationInfo        : 
    MyCommand        : Find-PSResource
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 139
    Line             : find-psresource -name Takeown -Repository PSGallery
    Statement        : find-psresource -name Takeown -Repository PSGallery
    PositionMessage  : At line:1 char:1
                       + find-psresource -name Takeown -Repository PSGallery
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : find-psresource
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo :
```

### Environment data

```powershell
PS> get-module *PSResourceGet; $PSVersionTable | ft

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     0.5.23     beta23     Microsoft.PowerShell.PSResourceGet  {Find-PSResource, Get-InstalledPSResource, Get-PSResourceRepository, Get-PSScri…


Name                           Value
----                           -----
PSVersion                      7.4.0-preview.3
PSEdition                      Core
GitCommitId                    7.4.0-preview.3
OS                             Darwin 22.5.0 Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:22 PDT 2023; root:xnu-8796.121.3~7/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by JamesWTruher** [comment]
additional information
probably something internal because starting a new shell does not exhibit the problem
```powershell
PS> set-psresourceRepository -name PSGallery -Trusted  
PS> find-psresource -name Takeown -Repository PSGallery
Find-PSResource: Repository 'PSGallery' is not a known repository type that is supported. Please file an issue for support at https://github.com/PowerShell/PSResourceGet/issues
PS> pwsh
PowerShell 7.3.4

   A new PowerShell stable release is available: v7.3.6 
   Upgrade now, or check out the release page at:       
     https://aka.ms/PowerShell-Release?tag=v7.3.6       

Loading personal and system profiles took 608ms.
🐚> import-module Microsoft.PowerShell.PSResourceGet
🐚> find-psresource -name Takeown -Repository PSGallery

Name    Version Prerelease Repository Description
----    ------- ---------- ---------- -----------
Takeown 1.0.2              PSGallery  Crescendo Powershell wrapper of takeown.exe
```

**Comment by SydneyhSmith** [comment]
closing as resolved
